clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1492,10 +1487,24 @@ namespace {
         EqualExprTy Vars1 = CollectVariableSet(S, E1);
         EqualExprTy Vars2 = CollectVariableSet(S, E2);
 
-        if (AddFreeVariables(S, Vars1, Vars2, EquivExprs, Pos1, FreeVars))
+        // EquivVars holds sets of DeclRefExpr and IntegerLiteral filtered from
+        // EquivExprs.
+        EquivExprSets EquivVars;
+        for (auto ExprSet : *EquivExprs) {
+          EqualExprTy Vars;
+          auto It = ExprSet.begin();
+          for (; It != ExprSet.end(); It++) {
+            *It = (*It)->IgnoreCasts();
```

Comment:
This also needs to ignore parentheses: 

```
Expr *E = (*It)->IgnoreParenCasts();
if (isa<IntegerLiteral>(E) || isa<DeclRefExpr>(E))
  Vars.push_back(E);
``` 

Otherwise, the following test case fails:

```
int a1 checked[5];
short int t20 : byte_count(5 * sizeof(int)) = ((((short int)(a1))));
```